utilities/transactions/transaction_db_impl.cc

```diff
@@ -447,6 +447,41 @@ TransactionLockMgr::LockStatusData TransactionDBImpl::GetLockStatusData() {
   return lock_mgr_.GetLockStatusData();
 }
 
+void DeadlockInfoBuffer::AddNewPath(std::vector<DeadlockInfo>& path) {
+  buffer_idx = (buffer_idx + 1) % n_latest_dlocks;
+  if (paths_buffer[buffer_idx].size()) {
+    paths_buffer[buffer_idx].clear();
+  }
+  paths_buffer[buffer_idx] = path;
+}
+
+std::vector<std::vector<DeadlockInfo>> DeadlockInfoBuffer::prepareBuffer() {
+  std::vector<std::vector<DeadlockInfo>> working;
+  for (uint32_t i = buffer_idx, counter = 0; counter < n_latest_dlocks;
+       counter++) {
+    if (!paths_buffer[i].size()) {
+      break;
+    }
+
+    working.push_back(paths_buffer[i]);
+    if (i == 0) {
+      i = n_latest_dlocks;
+    }
+    i = (i - 1) % n_latest_dlocks;
+  }
+  return working;
+}
+
+DeadlockInfoBuffer::DeadlockInfoBuffer(uint32_t latest_dlocks)
+    : n_latest_dlocks(latest_dlocks) {
+  paths_buffer = std::vector<std::vector<DeadlockInfo>>(n_latest_dlocks);
```

Comment:
I feel like n_latest_dlocks is redundant, since it's implicitly stored in the vector size.